### PR TITLE
Added the captain form for putting fuel to the boat

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

Added the captain form for putting fuel to the boat

[Task](https://www.notion.so/tianlu/Create-the-captain-form-for-putting-fuel-to-the-boat-5d50a53e03bd4634aaeea8a7cff6ed38?pvs=4)

## Test

- Clone this repo
- got to this URL http://localhost:<portnumber>/puttingFuel/fd300167b37f43e3befff48dcb16d711

## Platform

google chrome

## Proof of Concept
image

### CODE REVIEW

File: src/app/feedback/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import Sidebar from "@/components/sidebar/sidebar";
+import { getBookingInfo, getBoatInfo } from "@/services/notion.service";
+import { useEffect, useState } from "react";
+import router from "next/router";
+import FeedbackForm from "@/components/feedback";
+
+export default function UserFeedbackPage({ params }: { params: { id: string } }) {
+  const [boatInfo, setBoatInfo] = useState<any>({});
+  const [loading, setLoading] = useState<boolean>(true);
+  const [data, setData] = useState<any>({});
```
Comment (abelbordonado) [inline src/app/feedback/[id]/page.tsx:12]: Same comments

boat info is type boat
data here is the type `Fuel` (May neet to be added)

Actually here we don't load the boat info or the data info as it is a new record.

File: src/app/feedback/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      setData(data);
+      setLoading(false);
+    };
+
+    getBookingDetails();
+  }, [params.id]);
+
+  if (!data || !boatInfo) {
+    return;
+  }
```
Comment (abelbordonado) [inline src/app/feedback/[id]/page.tsx:39]: No needed

File: src/app/feedback/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <>
+      <section className="gradient-form justify-center h-screen w-full text-black">
+        <div className="g-6 flex h-full flex-wrap items-center justify-center text-neutral-800 dark:text-neutral-200">
+          <div className="flex md:flex-row flex-col w-full lg:flex lg:flex-wrap h-screen">
```
Comment (abelbordonado) [inline src/app/feedback/[id]/page.tsx:44]: I see this is in very page. Can we do a common component for it?
Or it shall go maybe on the layout

File: src/app/puttingFuel/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        return;
+      }
+
+      await getBoatDetails(data);
+      setData(data);
+      setLoading(false);
+    };
+
+    getBookingDetails();
+  }, [params.id]);
```
Comment (abelbordonado) [inline src/app/puttingFuel/[id]/page.tsx:35]: Probabbly no need in this form

File: src/app/puttingFuel/[id]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import Sidebar from "@/components/sidebar/sidebar";
+import { getBookingInfo, getBoatInfo } from "@/services/notion.service";
+import { useEffect, useState } from "react";
+import router from "next/router";
+import PuttingFuelForm from "@/components/puttingFuel";
+
+export default function PuttingFuel({ params }: { params: { id: string } }) {
```
Comment (abelbordonado) [inline src/app/puttingFuel/[id]/page.tsx:9]: Lets call it just `fuel`

File: src/components/puttingFuel/index.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+};
+
+export default function PuttingFuelForm({
+  data,
+  setData,
+  boatInfo,
+}: {
+  data: any;
+  setData: any;
+  boatInfo: Boat;
```
Comment (abelbordonado) [inline src/components/puttingFuel/index.tsx:28]: Probably not needed

File: src/components/puttingFuel/index.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+              <CommonSelect
+                id="boats"
+                name="boats"
+                data={SEABOB} //Abel this one I will change in integration
+                value={data["Fuel Payment"]}
+                onChange={(e) =>
+                  setData({ ...data, "Fuel Payment": e.target.value })
+                }
+                required
+              />
```
Comment (abelbordonado) [inline src/components/puttingFuel/index.tsx:70]: In my other branch there is the notion function to return all the boats and captains, we can load on the component it self and create a 

<SelectCaptain>. and <SelectBoat> input type

**Comment by abelbordonado** [comment]
image

File: src/components/puttingFuel/index.tsx (hunk 7 above)
Comment (kayiganwa04) [reply src/components/puttingFuel/index.tsx:70]: checking

**Comment by abelbordonado** [comment]
image
We got 3 times other lol

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Pull request merged**